src/Cli/dotnet/commands/dotnet-sln/add/Program.cs

```diff
@@ -2,143 +2,180 @@
+        public override int Execute()
+        {
+            if (_projects.Count == 0)
+            {
+                throw new GracefulException(CommonLocalizableStrings.SpecifyAtLeastOneProjectToAdd);
+            }
+            string solutionFileFullPath = SlnCommandParser.GetSlnFileFullPath(_fileOrDirectory);
 
-            if (hasRelativeRoot)
+            try
             {
-                relativeRoot = PathUtility.GetPathWithDirectorySeparator(relativeRoot);
-                _relativeRootSolutionFolders = relativeRoot.Split(Path.DirectorySeparatorChar, StringSplitOptions.RemoveEmptyEntries);
+                PathUtility.EnsureAllPathsExist(_projects, CommonLocalizableStrings.CouldNotFindProjectOrDirectory, true);
+                IEnumerable<string> fullProjectPaths = _projects.Select(project =>
+                {
+                    var fullPath = Path.GetFullPath(project);
+                    return Directory.Exists(fullPath) ? MsbuildProject.GetProjectFileFromDirectory(fullPath).FullName : fullPath;
+                });
+                AddProjectsToSolutionAsync(solutionFileFullPath, fullProjectPaths, CancellationToken.None).Wait();
+                return 0;
             }
-            else
+            catch (Exception ex) when (ex is not GracefulException)
             {
-                _relativeRootSolutionFolders = null;
+                {
+                    if (ex is SolutionException || ex.InnerException is SolutionException)
+                    {
+                        throw new GracefulException(CommonLocalizableStrings.InvalidSolutionFormatString, solutionFileFullPath, ex.Message);
+                    }
+                    throw new GracefulException(ex.Message, ex);
+                }
             }
         }
 
-        public override int Execute()
+        private async Task AddProjectsToSolutionAsync(string solutionFileFullPath, IEnumerable<string> projectPaths, CancellationToken cancellationToken)
         {
-            SlnFile slnFile = SlnFileFactory.CreateFromFileOrDirectory(_fileOrDirectory);
-
-            var arguments = (_parseResult.GetValue<IEnumerable<string>>(SlnAddParser.ProjectPathArgument) ?? Array.Empty<string>()).ToList().AsReadOnly();
-            if (arguments.Count == 0)
+            ISolutionSerializer serializer = SlnCommandParser.GetSolutionSerializer(solutionFileFullPath);
+            SolutionModel solution = await serializer.OpenAsync(solutionFileFullPath, cancellationToken);
+            // set UTF8 BOM encoding for .sln
+            if (serializer is ISolutionSerializer<SlnV12SerializerSettings> v12Serializer)
             {
-                throw new GracefulException(CommonLocalizableStrings.SpecifyAtLeastOneProjectToAdd);
+                solution.SerializerExtension = v12Serializer.CreateModelExtension(new()
+                {
+                    Encoding = new UTF8Encoding(encoderShouldEmitUTF8Identifier: true)
+                });
             }
```

Comment:
Is this the expected behavior? With the old code, I tried to do dotnet sln add on a project that was a release project where the solution only had a debug configuration (I removed the Release line) and it did not add a Release line to the solution.

Reply:
You mean the default configurations or the encoding? If it is the former, yes, there is a test that explicitly expects UTF8 encoding fo sln files. If it is the latter, I can try experimenting a bit with this, as it indeed seems weird

Reply:
It's about the configurations!

Reply:
⭐  I think this is the only important remaining thing now.. This is more about the 
` // Set default configurations and platforms for sln file` line. 

Basically, when I had tested it: if you do `dotnet new console -n release`, and `dotnet new sln`, then remove the release line from the sln and `dotnet sln add release`, I think it didn't used to have a release line after add, but now I think it does. Is that ok? ... Maybe? It is a 'breaking change' if you can replicate that behavior. I would consult with the sln team if so on if this is ok.

Reply:
Oh. Removing this does cause some tests to break. But after analyzing it further, it might be by how the sln files it compares them to are generated. I will try replicating it and see what it generates :)

Reply:
I looked into this and this is required for some tests to pass but we could potentially bring this to discussion if we want to add arm64 support by default @nagilson @baronfel

Reply:
The current behavior also adds these default configurations, https://github.com/dotnet/sdk/blob/10d457d2b369ea1fbcb8c3f6d02bd43e08411146/src/Cli/dotnet/SlnFileExtensions.cs#L128
So this shouldn't be a breaking issue :)